File: Cocktail/src/main/kotlin/org/programmers/cocktail/search/service/CommentsService.java
package org.programmers.cocktail.search.service;

import java.util.Collections;
import java.util.List;
import org.programmers.cocktail.entity.Comments;
import org.programmers.cocktail.repository.comments.CommentsRepository;
import org.programmers.cocktail.search.dto.CommentsTO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class CommentsService {

    static final int SUCCESS = 1;
    static final int FAIL = 0;

    @Autowired
    CommentsRepository commentsRepository;

    @Autowired
    CommentsMapper commentsMapper;

    public List<CommentsTO> findByCocktailId(Long cocktailId){

        List<Comments> commentsList = commentsRepository.findByCocktailId(cocktailId);

        if(commentsList.isEmpty()){
            return Collections.emptyList();
        }

        List<CommentsTO> commentsTOList = commentsMapper.convertToCommentsTOList(commentsList);

        return commentsTOList;
    }

    public int insertComments(CommentsTO commentsTO){

        // TO->Entity 변환
        Comments comments = commentsMapper.convertToComments(commentsTO);
        try {
            commentsRepository.save(comments);
        } catch (Exception e) {
            System.out.println("[에러]"+e.getMessage());
            return FAIL;
        }

        return SUCCESS;
    }

    public int deleteById(CommentsTO commentsTO){

        Long commentsId = commentsTO.getId();

        int commentsDeleteResult = commentsRepository.deleteByIdWithReturnAffectedRowCount(commentsId);

        if(commentsDeleteResult == 0){
            return FAIL;
        }

        return SUCCESS;
    }


}

File: Cocktail/src/main/kotlin/org/programmers/cocktail/search/service/CommentsService.kt
package org.programmers.cocktail.search.service

import org.programmers.cocktail.repository.comments.CommentsRepository
import org.programmers.cocktail.search.dto.CommentsTO
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class CommentsService {
    @Autowired
    var commentsRepository: CommentsRepository? = null

    @Autowired
    var commentsMapper: CommentsMapper? = null

    fun findByCocktailId(cocktailId: Long?): List<CommentsTO> {
        val commentsList = commentsRepository!!.findByCocktailId(cocktailId)

        if (commentsList!!.isEmpty()) {
            return emptyList()
        }

        val commentsTOList = commentsMapper!!.convertToCommentsTOList(commentsList)

        return commentsTOList
    }

    fun insertComments(commentsTO: CommentsTO?): Int {
        // TO->Entity 변환

        val comments = commentsMapper!!.convertToComments(commentsTO)
        try {
            commentsRepository!!.save(comments)
        } catch (e: Exception) {
            println("[에러]" + e.message)
            return FAIL
        }

        return SUCCESS
    }

    fun deleteById(commentsTO: CommentsTO): Int {
        val commentsId: Long = commentsTO.setId()

        val commentsDeleteResult =
            commentsRepository!!.deleteByIdWithReturnAffectedRowCount(commentsId)

        if (commentsDeleteResult == 0) {
            return FAIL
        }

        return SUCCESS
    }


    companion object {
        const val SUCCESS: Int = 1
        const val FAIL: Int = 0
    }
}
